packages/zwave-js/src/lib/driver/Driver.ts

```diff
@@ -2612,6 +2605,12 @@ ${handlers.length} left`,
 				);
 				return;
 			}
+
+			if (unwrapped instanceof SupervisionCCGet) {
+				msg.supervisionEncapsulation = unwrapped;
+			}
+
+			msg.encapsulatingCC.push(msg.command);
```

Comment:
IMO tracking the encapsulation stack should be part of the while loop above. Not sure if we need the special case for `supervisionEncapsulation` or if it could be done with a check on the encapsulation array.

Reply:
See https://github.com/zwave-js/node-zwave-js/pull/3240#discussion_r702424342 - I don't think we need to add any additional tracking - it seems the existing constructor preserves the encapsulation chain.